docs/developers-guide/build.md

```diff
@@ -1,25 +1,132 @@
+This doc will show you how you can build and run Metabase on your own computer so you can play around with it as you please or test features that are in development. You can also run development branches of Metabase [using a pre-built Docker image](dev-branch-docker.md).
 
-These are the tools which are required in order to complete any build of the Metabase code. Follow the links to download and install them on your own before continuing.
+## Install the prerequisites
+
+If you're using macOS, you'll want to install Xcode Command Line Tools first, by running: 
+
+```
+xcode-select --install
+```
+
+And here are the tools you'll need to complete any build of the Metabase code. Follow the links to download and install them before continuing.
 
 1. [Clojure (https://clojure.org)](https://clojure.org/guides/getting_started) - install the latest release by following the guide depending on your OS
+
 2. [Java Development Kit JDK (https://adoptopenjdk.net/releases.html)](https://adoptopenjdk.net/releases.html) - you need to install JDK 11 ([more info on Java versions](../operations-guide/java-versions.md))
+
 3. [Node.js (http://nodejs.org/)](http://nodejs.org/) - latest LTS release
-4. [Yarn package manager for Node.js](https://yarnpkg.com/) - latest release of version 1.x - you can install it in any OS by doing `npm install --global yarn`
+
+4. [Yarn package manager for Node.js](https://yarnpkg.com/) - latest release of version 1.x - you can install it in any OS by running:
+ 
+```
+npm install --global yarn
+```
 
 On a most recent stable Ubuntu/Debian, all the tools above, with the exception of Clojure, can be installed by using:
 
 ```
 sudo apt install openjdk-11-jdk nodejs && sudo npm install --global yarn
 ```
-If you have multiple JDK versions installed in your machine, be sure to switch your JDK before building by doing `sudo update-alternatives --config java` and selecting Java 11 in the menu
+
+If you have multiple JDK versions installed in your machine, be sure to switch your JDK before building with: 
+
+```
+sudo update-alternatives --config java
+``` 
+
+Then select Java 11 in the menu.
 
 If you are developing on Windows, make sure to use Ubuntu on Windows and follow instructions for Ubuntu/Linux instead of installing ordinary Windows versions.
 
 Alternatively, without the need to explicitly install the above dependencies, follow the guide [on using Visual Studio Code](deven.md#developing-with-visual-studio-code.md) and its remote container support.
 
+## Copy the Metabase code to your computer
+
+Once you have all that installed, you’re going to get a copy of Metabase's source code from GitHub, where it’s stored.
+
+1. Create a `workspace` folder (you can name it that or whatever you want). You’ll put the Metabase code files inside this `workspace` folder.
+
+2. Open up your terminal app, and navigate to your workspace folder with: 
+
+```
+cd ~/workspace
+```
+
+1. Run this GitHub command to “clone” Metabase into this folder
```

Comment:
Do they need to set up the `gh` CLI to do this? Should that be in the required tools?

https://cli.github.com/manual/

And should we provide the alternate git clone command?